api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -93,6 +93,14 @@ type EnvoyProxyKubernetesProvider struct {
 	//
 	// +optional
 	EnvoyService *KubernetesServiceSpec `json:"envoyService,omitempty"`
+
+	// TLSConfigSecret is a reference to the secret containing the TLS certificate,
+	// certificate authority certificate, and private key used to secure Envoy's
+	// client connection to the control plane. Defaults to {Name: "envoy",
+	// Certificate: "tls.crt", PrivateKey: "tls.key", CACertificate: "ca.crt"}.
+	//
+	// +optional
+	TLSConfigSecret *KubernetesTLSConfigSecret `json:"tlsConfigSecret,omitempty"`
```

Comment:
I considered putting this in `EnvoyProxyKubernetesProvider.EnvoyDeployment  *KubernetesDeploymentSpec` but this is shared with `EnvoyGatewayKubernetesProvider.RateLimitDeployment` so I wasn't certain if it made sense for both to have the option to configure a `KubernetesTLSConfigSecret`.

Reply:
for the name of the field, I vote for `controlPlaneAuth` or something that keeps it similar to what is defined in `EnvoyGateway`
reg the datatype https://github.com/envoyproxy/gateway/pull/1411#discussion_r1189172867 should be relevant